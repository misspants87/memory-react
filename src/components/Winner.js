import React from 'react';

const Winner = props => {

  return (
    <div className="winner">
      {`The winner is ${props.winner}`}
    </div>
  )
}

export default Winner;
