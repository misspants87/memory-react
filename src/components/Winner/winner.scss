.winner {
  display: flex;
  flex-direction: column;
  // justify-content: flex-end;
  align-items: center;
  font-family: 'Roboto';
  font-size: 20pt;
  color: white;

  button {
    background-color: white;
    color: black;
    border-radius: 5px;
  }
}
