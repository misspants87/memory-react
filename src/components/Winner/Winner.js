import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import './winner.scss';
import url from '../../assets/party-horn.mp3';

const Winner = props => {

  let audio = new Audio(url);
//allows access to methods in confetti cdn
  let confetti = window.confetti;
  console.log(confetti);
  useEffect(() => {
    confetti({
      particleCount: 150,
      spread: 180,
    });
    // confetti.start();
    audio.play();
    // return () => {
    //   confetti.stop();
    // }
  }, [])

  return (
    <div className="winner" data-test="Winner Component">
      <h4>
      {`The winner is ${props.winner}`}
      </h4>
    </div>
  )
}

Winner.propTypes = {
  winner: PropTypes.string,
}

export default Winner;
