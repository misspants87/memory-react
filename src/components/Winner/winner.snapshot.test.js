import React from 'react';
import renderer from 'react-test-renderer';
import Winner from './Winner';

test('should render Winner Component correctly', () => {
  let component = renderer.create(
    <Winner
      winner = {'me'}
    />
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
