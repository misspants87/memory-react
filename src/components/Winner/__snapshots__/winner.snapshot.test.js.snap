// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Winner Component correctly 1`] = `
<div
  className="winner"
>
  The winner is me
</div>
`;
